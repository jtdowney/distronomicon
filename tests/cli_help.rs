use assert_cmd::cargo::cargo_bin_cmd;
use insta::assert_snapshot;

#[test]
fn test_help_output() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}

#[test]
fn test_check_subcommand_help() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("check").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}

#[test]
fn test_update_subcommand_help() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("update").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}

#[test]
fn test_version_subcommand_help() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("version").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}
