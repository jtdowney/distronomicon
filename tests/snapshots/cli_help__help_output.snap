---
source: tests/cli_help.rs
expression: stdout
---
Usage: distronomicon [OPTIONS] --app <APP> <COMMAND>

Commands:
  check    Check for updates without installing
  update   Update to the latest release
  version  Show currently installed version
  help     Print this message or the help of the given subcommand(s)

Options:
      --repo <REPO>
          GitHub repository (owner/name)
      --app <APP>
          Application name
      --pattern <PATTERN>
          Asset filename pattern (regex)
      --checksum-pattern <CHECKSUM_PATTERN>
          Checksum filename pattern (optional)
      --allow-prerelease
          Allow prerelease versions
      --github-token <GITHUB_TOKEN>
          GitHub API token [env: GITHUB_TOKEN]
      --github-host <GITHUB_HOST>
          GitHub API host [env: GITHUB_HOST=] [default: api.github.com]
      --restart-command <RESTART_COMMAND>
          Command to run after successful update
      --retain <RETAIN>
          Number of releases to retain [default: 3]
      --skip-verification
          Skip checksum verification
      --state-directory <STATE_DIRECTORY>
          State directory [env: STATE_DIRECTORY=]
      --install-root <INSTALL_ROOT>
          Install root directory [env: PREFIX=] [default: /opt]
  -v, --verbose...
          Verbose output (-v, -vv)
  -h, --help
          Print help
